webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala

```diff
@@ -3068,11 +3068,15 @@ class StringFormatter private (val maybeSettings: Option[KnoraSettingsImpl] = No
     * Creates a new value IRI based on a UUID.
     *
     * @param resourceIri the IRI of the resource that will contain the value.
+    * @param givenUUID   the optional given UUID of the value. If not provided, create a random one.
     * @return a new value IRI.
     */
-  def makeRandomValueIri(resourceIri: IRI): IRI = {
-    val knoraValueUuid = makeRandomBase64EncodedUuid
-    s"$resourceIri/values/$knoraValueUuid"
+  def makeRandomValueIri(resourceIri: IRI, givenUUID: Option[UUID] = None): IRI = {
```

Comment:
Do we check somewhere that the UUID is a V4 UUID?

Reply:
each given UUID is a 22-character string. Then the given UUID or randomly created one is encoded to base64 UUID of V4 type. Any invalid custom UUID fails in the encoding process and an Illegal Argument exception is thrown.

Reply:
thanks